Tests/UpstreamIntegrationTests/AnyCodableAssertsParameterizedTests.swift

```diff
@@ -0,0 +1,420 @@
+                    assertExactMatch(expected: expected, actual: actual)
+                }
+
+                assertTypeMatch(expected: expected, actual: actual)
+            }
+        }
+    }
+    
+    func testFlexibleCollectionTypeMatching() {
+        let rawCases: [(expected: String, actual: String)] = [
+            (expected: #"[]"#, actual: #"[1]"#),
+            (expected: #"[1,2,3]"#, actual: #"[1,2,3,4]"#),
+            (expected: #"{}"#, actual: #"{ "k": "v" }"#),
+            (expected: #"{ "key1": 1, "key2": "a", "key3": 1.0, "key4": true }"#,
+             actual: #"{ "key1": 1, "key2": "a", "key3": 1.0, "key4": true, "key5": "extra" }"#),
+        ]
+        let testCases = rawCases.map { tuple in
+            return (expected: getAnyCodable(tuple.expected)!, actual: getAnyCodable(tuple.actual))
+        }
+        for (index, (expected, actual)) in testCases.enumerated() {
+            XCTContext.runActivity(named: "should pass flexible matching when expected is a subset: [\(index)]: test with expected=\(expected), actual=\(actual)") { activity in
+                XCTExpectFailure("Validation should fail when asserting exact equality for collections where the actual has extra elements or keys") {
+                    assertEqual(expected: expected, actual: actual)
+                }
+                assertExactMatch(expected: expected, actual: actual)
+                assertTypeMatch(expected: expected, actual: actual)
+            }
+        }
+    }
+    
+    func testFailure() {
+        let rawCases: [(expected: Any?, actual: Any?)] = [
+            (expected: 1, actual: 2.0), // [0]
+            (expected: 1, actual: 1.0), // Tests Int vs Double type detection
+            (expected: 1, actual: "a"),
+            (expected: 1, actual: true),
+            (expected: 1, actual: [:] as [String: Any]),
+            (expected: 1, actual: [] as [Any]), // [5]
+            (expected: 1, actual: nil),
+            (expected: 2.0, actual: 1),
+            (expected: 2.0, actual: 2), // Tests Double vs Int type detection
+            (expected: 2.0, actual: "a"),
+            (expected: 2.0, actual: true), // [10]
+            (expected: 2.0, actual: [:] as [String: Any]),
+            (expected: 2.0, actual: [] as [Any]),
+            (expected: 2.0, actual: nil),
+            (expected: "a", actual: 1),
+            (expected: "a", actual: 2.0), // [15]
+            (expected: "a", actual: true),
+            (expected: "a", actual: [:] as [String: Any]),
+            (expected: "a", actual: [] as [Any]),
+            (expected: "a", actual: nil),
+            (expected: true, actual: 1), // [20]
+            (expected: true, actual: 2.0),
+            (expected: true, actual: "a"),
+            (expected: true, actual: [:] as [String: Any]),
+            (expected: true, actual: [] as [Any]),
+            (expected: true, actual: nil),
+            (expected: false, actual: 1), // [25]
```

Comment:
Let's update (expected: false, actual: 1) to (expected: false, actual: 0) since 0 can be treated as false. So just verifying no intelligent casting.